### PR TITLE
[MAINTENANCE] Separate docs_integrations tests by backend

### PR DESCRIPTION
**Pull request by anthonyburdi**
Changes proposed in this pull request:
- Split up tests in CI by engine / backend. Parametrization coming in a future PR (https://learn.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops).
- Start moving IntegrationTestFixtures into a new `test_definitions` folder instead of under `db`. I think this can be carried further a bit in upcoming PRs, we need to split up more clearly: data used in tests, "fixtures" or gx configurations, test definitions, actual test scripts.
- Add Athena, Redshift and a few other missing engines/backends to the stage.

### Definition of Done

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

**Comment by anthonyburdi** [comment]
Closing for now to not burn too many CI resources.

File: ci/azure-pipelines-docs-integration.yml [hunk 1]
```diff
@@ -97,28 +97,120 @@ stages:
     pool:
       vmImage: 'ubuntu-latest'
     jobs:
-    - job: test_docs_not_mysql
-      timeoutInMinutes: 120
+    - job: test_docs_non_backend_specific
```
Comment (Shinnnyshinshin) [inline ci/azure-pipelines-docs-integration.yml:100]: Much better name :)

File: tests/integration/backend_dependencies.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
 class BackendDependencies(enum.Enum):
     AWS = "AWS"
+    ATHENA = "ATHENA"
```
Comment (Shinnnyshinshin) [inline tests/integration/backend_dependencies.py:6]: Thank you for this too. A bit surprised it wasn't included already ..

File: tests/integration/test_script_runner.py [hunk 3]
```diff
@@ -1807,3 +1591,5 @@ def _check_for_skipped_tests(pytest_args, integration_test_fixture) -> None:
         pytest.skip("Skipping Azure tests")
     elif BackendDependencies.TRINO in dependencies and not pytest_args.trino:
         pytest.skip("Skipping Trino tests")
+    elif BackendDependencies.ATHENA in dependencies and not pytest_args.athena:
+        pytest.skip("Skipping Athena tests")
```
Comment (Shinnnyshinshin) [inline tests/integration/test_script_runner.py:1595]: ❤️

**Review by Shinnnyshinshin — APPROVED** [review]
Looks great @anthonyburdi thank you 🙇🏼

**Pull request merged**